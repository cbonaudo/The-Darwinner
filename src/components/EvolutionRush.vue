<template>
  <div class="rush" :style="getPosition" v-if="isDisplayed"></div>
</template>

<script>
export default {
  name: 'EvolutionRush',
  data() {
    return { top: 5, left: 5 }
  },
  props: {
    isDisplayed: { default: true }
  },
  computed: {
    getPosition() {
      return { top: `${this.top}%`, left: `${this.left}%` }
    }
  },
  methods: {
    changePosition() {
      this.top = Math.random() * 65 + 5
      this.left = Math.random() * 75 + 5
    }
  }
}
</script>

<style scoped>
.rush {
  background-color: black;
  top: 5%;
  height: 200px;
  width: 200px;
  border-radius: 5%;
  position: fixed;
}
</style>
