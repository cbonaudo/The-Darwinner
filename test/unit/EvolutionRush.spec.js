import EvolutionRush from '@/components/EvolutionRush'
import { shallowMount } from '@vue/test-utils'

describe('EvolutionRush.vue', () => {
  const wrapper = shallowMount(EvolutionRush)

  describe('isDisplayed', () => {
    it('should render if displayed', () => {
      wrapper.setProps({ isDisplayed: false })
      expect(wrapper.element).toMatchSnapshot()
    })
    it('shouldnâ€™t render if not displayed', () => {
      wrapper.setProps({ isDisplayed: true })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  it('should have correct starting data', () => {
    expect(EvolutionRush.data()).toMatchSnapshot()
  })

  describe('changePosition', () => {
    it('creates a new position', () => {
      wrapper.setData({ top: '5%', left: '5%' })
      expect(wrapper.vm.top).toEqual('5%')
      expect(wrapper.vm.left).toEqual('5%')
      wrapper.vm.changePosition()
      expect(wrapper.vm.top !== '5%').toBeTruthy()
      expect(wrapper.vm.left !== '5%').toBeTruthy()
    })
  })

  describe('getPosition', () => {
    it('returns the position object', () => {
      expect(EvolutionRush.computed.getPosition.call({ top: 5, left: 5 })).toEqual({ top: '5%', left: '5%' })
    })
    it('returns the position object', () => {
      expect(EvolutionRush.computed.getPosition.call({ top: 12.5, left: 12.5 })).toEqual({
        top: '12.5%',
        left: '12.5%'
      })
    })
  })
})
