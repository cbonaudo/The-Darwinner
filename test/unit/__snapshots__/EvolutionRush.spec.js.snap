// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvolutionRush.vue isDisplayed should render if displayed 1`] = `
<div
  class="rush"
  style="top: 5%; left: 5%;"
/>
`;

exports[`EvolutionRush.vue isDisplayed shouldnâ€™t render if not displayed 1`] = `<!---->`;

exports[`EvolutionRush.vue should have correct starting data 1`] = `
Object {
  "left": 5,
  "top": 5,
}
`;
